src/internal/pool/pool.go

```diff
@@ -40,7 +40,7 @@ type Pool struct {
 func NewPool(kubeClient *kube.Clientset, namespace string, serviceName string, port int, queueSize int64, opts ...grpc.DialOption) (*Pool, error) {
 	endpointsInterface := kubeClient.CoreV1().Endpoints(namespace)
 
-	watch, err := endpointsInterface.Watch(metav1.ListOptions{
+	watch, err := endpointsInterface.Watch(context.TODO(), metav1.ListOptions{
```

Comment:
I think this could be a `context.Background` since the watch is cancelled when the connection is closed below.  But TODO is fine for now.